src/Nerdbank.MessagePack/MessagePackSerializer.cs

```diff
@@ -77,6 +78,11 @@ public record MessagePackSerializer
 	/// </summary>
 	public SerializationContext StartingContext { get; init; } = new();
 
+	/// <summary>
+	/// Gets the shape providers that may be called upon for types that must be serialized at runtime.
+	/// </summary>
+	public ImmutableArray<ITypeShapeProvider> ShapeProviders { get; init; } = [];
```

Comment:
@eiriktsarpalis  What can I provide here as a default so that `[GenerateShape]`-attributed types will be found?
I suppose folks should have to opt in for reflection-based. But I'd like to enable types with that attribute to be automatically discovered if possible. Do I have to expose methods for those cases with generic type constraints to ensure that they implement `IShapeable<T>` in order for this to work?

Reply:
I went with another strong-typed approach, which requires that the converter itself provide the type shape (via an `IShapeable<T>` type constraint).

Reply:
But I *am* curious whether there's a use case for a `ImmutableArray<ITypeShapeProvider> ShapeProviders` member on the serializer. Maybe for the .NET Framework build where `IShapeable<T>` doesn't exist?